### PR TITLE
Support for specifying a filename

### PR DESCRIPTION
**Pull request by Ricoch**
*Note*
- Since this change is enough to release an initial version of the gem, I shifted the version to 1.0.0, even though I'm not sure what version number should we start with, so, any comments regarding this so we can discuss are welcome.

Description:
- Added support for when a user wants to specify a filename for their attachment
- Added tests for filename
- Renamed method `attachment_from_base64` to `attachment_from_data`, since data may or may not be a base64 encoded string.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -109,6 +109,25 @@ class UsersController < ApplicationController
 end
 ```
 
+## Specifying a filename or content_type
+
+If you are willing to add a specific filename to your attachment, or send in a specific content_type for your file, you can use `data:` to attach the base64 data and specify your `filename:` and/or `content_type:` hash keys afterwards.
```
Comment (santib) [inline README.md]: why `afterwards`?

File: lib/active_storage_support/base64_attach.rb [hunk 2]
```diff
@@ -3,14 +3,33 @@ module ActiveStorageSupport
   module Base64Attach
     module_function
 
-    def attachment_from_base64(data:, filename: Time.current.to_i.to_s)
-      return data unless data.is_a?(String) && data.strip.start_with?('data')
+    def attachment_from_data(data:, filename: Time.current.to_i.to_s, content_type: nil)
+      case data
+      when Hash
+        return data unless data.key?(:data)
+        filename = data[:filename] if data.key?(:filename)
```
Comment (santib) [inline lib/active_storage_support/base64_attach.rb]: shouldn't you be using `filename` here? Why `data[:filename]`?

File: lib/active_storage_support/base64_attach.rb [hunk 3]
```diff
@@ -3,14 +3,33 @@ module ActiveStorageSupport
   module Base64Attach
     module_function
 
-    def attachment_from_base64(data:, filename: Time.current.to_i.to_s)
-      return data unless data.is_a?(String) && data.strip.start_with?('data')
+    def attachment_from_data(data:, filename: Time.current.to_i.to_s, content_type: nil)
+      case data
+      when Hash
```
Comment (santib) [inline lib/active_storage_support/base64_attach.rb]: why all these changes? wasn't it working just returning `data` if it was not a string?

File: active_storage_base64.gemspec [hunk 4]
```diff
@@ -1,13 +1,14 @@
 Gem::Specification.new do |s|
   s.name = 'active_storage_base64'
-  s.version = '0.0.0'
+  s.version = '1.0.0'
```
Comment (santib) [inline active_storage_base64.gemspec]: Following https://semver.org/#how-should-i-deal-with-revisions-in-the-0yz-initial-development-phase maybe we can start with `0.1.0` until we have some apps in prod using the gem?

File: README.md (hunk 1 above)
Comment (Ricoch) [reply README.md]: True, it doesn't need an specific order. Removing.

File: lib/active_storage_support/base64_attach.rb (hunk 3 above)
Comment (santib) [reply lib/active_storage_support/base64_attach.rb]: But you are not supporting `user.avatar.attach(params[:avatar])`. You are supporting `user.avatar.attach(data: params[:avatar])`. But why would you allow that? I don't think it adds any kind of value, keywords arguments are already a `Hash` so when you call `user.avatar.attach(data: params[:avatar], filename: 'filename')` this is exactly the same as `user.avatar.attach({ data: params[:avatar], filename: 'filename' })`.

File: README.md [hunk 5]
```diff
@@ -109,6 +109,42 @@ class UsersController < ApplicationController
...
+```ruby
+class UsersController < ApplicationController
+  def create
+    user = User.create(user_params)
+  end
+
+  private
+
+  def user_params
+    params.require(:user).permit(:username, :email, avatar:[:data,
```
[the remark this answers is not in the document]
Comment (Ricoch) [reply README.md]: Actually, yes, I should change that since we no longer support the attachment of only the base64 without specifying the data first.

**Pull request merged**